xarray/core/dataset.py

```diff
@@ -6864,6 +6877,174 @@ def bfill(self, dim: Hashable, limit: int | None = None) -> Self:
         new = _apply_over_vars_with_dim(bfill, self, dim=dim, limit=limit)
         return new
 
+    def fill_gaps(
+        self,
+        dim: Hashable,
+        use_coordinate: bool | Hashable = True,
+        limit: (
+            None
+            | int
+            | float
+            | str
+            | pd.Timedelta
+            | np.timedelta64
+            | datetime.timedelta
+        ) = None,
+        limit_direction: LimitDirectionOptions = "both",
+        limit_area: LimitAreaOptions | None = None,
+        max_gap: (
+            None
+            | int
+            | float
+            | str
+            | pd.Timedelta
+            | np.timedelta64
+            | datetime.timedelta
+        ) = None,
+    ) -> GapMask:
+        """Fill in gaps in the data using one of several filling methods.
+        Allows for fine control on how far to extend the valid data into the gaps and the maximum size of the gaps to fill.
```

Comment:
Could we clearly list the methods here on the returned object? (Or lmk if I'm missing them — they're in the doctest but otherwise I don't see them, thanks)